### PR TITLE
copr: vectorize make_set

### PR DESCRIPTION
**Pull request by trabbart**
Signed-off-by: Bartosz Podkanowicz <podkanowicz.bartosz@gmail.com>

### What problem does this PR solve?

vectorize function make_set

### Check List #5751 

Tests 

- Unit test

### CODE REVIEW

File: components/tidb_query_vec_expr/src/impl_string.rs [hunk 1]
```diff
@@ -1944,6 +1979,112 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_make_set() {
```
Comment (sticnarf) [inline components/tidb_query_vec_expr/src/impl_string.rs:1983]: Could you also test the case when the first parameter is `NULL`?

File: components/tidb_query_vec_expr/src/impl_string.rs [hunk 2]
```diff
@@ -1944,6 +1979,112 @@ mod tests {
...
+                vec![
+                    Some(1).into(),
+                    Some(b"DataBase".to_vec()).into(),
+                    Some(b"Hello World!".to_vec()).into(),
+                ],
+                Some(b"DataBase".to_vec()),
+            ),
+            (
+                vec![
+                    None::<Int>.into(),
```
Comment (trabbart) [inline components/tidb_query_vec_expr/src/impl_string.rs:2003]: This test case have first parameter equal to NULL.

File: components/tidb_query_vec_expr/src/impl_string.rs (hunk 1 above)
Comment (trabbart) [reply components/tidb_query_vec_expr/src/impl_string.rs:1983]: Third test case have the first parameter equal to `NULL`

Comment (sticnarf) [reply components/tidb_query_vec_expr/src/impl_string.rs:1983]: Ah sorry, I miss it.

File: components/tidb_query_vec_expr/src/impl_string.rs [hunk 3]
```diff
@@ -1944,6 +1979,112 @@ mod tests {
...
+                vec![
+                    Some(1).into(),
+                    None::<Bytes>.into(),
+                    Some(b"Hello World!".to_vec()).into(),
+                ],
+                Some(b"".to_vec()),
+            ),
+            (
+                vec![
+                    Some(3).into(),
```
Comment (zhongzc) [inline components/tidb_query_vec_expr/src/impl_string.rs:2020]: How about using Hex or Bin instead of Dec?

File: components/tidb_query_vec_expr/src/impl_string.rs [hunk 4]
```diff
@@ -1944,6 +1979,112 @@ mod tests {
...
+                ],
+                Some(b"Hello World!".to_vec()),
+            ),
+            (
+                vec![
+                    Some(0b10).into(),
+                    Some(b"DataBase".to_vec()).into(),
+                    Some(b"Hello World!".to_vec()).into(),
+                ],
+                Some(b"Hello World!".to_vec()),
```
Comment (zhongzc) [inline components/tidb_query_vec_expr/src/impl_string.rs:2049]: I suggest adding these cases:
```
make_set(0b100, "DataBase", "Hello World!")
make_set(0b110, "DataBase", "Hello World!")
```

File: components/tidb_query_vec_expr/src/impl_string.rs [hunk 5]
```diff
@@ -1944,6 +1979,112 @@ mod tests {
...
+                    Some(-2).into(),
+                    Some(b"a".to_vec()).into(),
+                    Some(b"b".to_vec()).into(),
+                    Some(b"c".to_vec()).into(),
+                ],
+                Some(b"b,c".to_vec()),
+            ),
+            (
+                vec![
+                    Some(-3).into(),
```
Comment (zhongzc) [inline components/tidb_query_vec_expr/src/impl_string.rs:2071]: We can format `-1` to `0xffffffff`, `-2` to `0xfffffffe`, `-3` to `0xfffffffd` to indicate they're bitmasks.

**Comment by zhongzc** [comment]
Please fix the lint. Rest LGTM. @trabbart

**Comment by trabbart** [comment]
@zhongzc I fixed lint. Thank you for review.
